lib/api/CAnomalyJob.cc

```diff
@@ -1051,15 +1055,26 @@ bool CAnomalyJob::backgroundPersistState(CBackgroundPersister& backgroundPersist
     std::sort(copiedDetectors.begin(), copiedDetectors.end(),
               maths::COrderings::SFirstLess());
 
-    if (backgroundPersister.addPersistFunc(std::bind(
+    if (m_PeriodicPersister->addPersistFunc(std::bind(
             &CAnomalyJob::runBackgroundPersist, this, args, std::placeholders::_1)) == false) {
         LOG_ERROR(<< "Failed to add anomaly detector background persistence function");
         return false;
     }
 
+    m_PeriodicPersister->useBackgroundPersistence();
+
     return true;
 }
 
+bool CAnomalyJob::runForegroundPersist(core::CDataAdder& persister) {
+    LOG_INFO(<< "Foreground persist commencing...");
+
+    // Prune the models so that the persisted state is as neat as possible
+    this->pruneAllModels();
+
+    return this->persistState(persister, "Periodic foreground persist at ");
```

Comment:
This class has a lot of persist methods: `persistState` two overloads, `runForegroundPersist`, `runBackgroundPersist`, `periodicPersistStateInBackground` and `periodicPersistStateInBackground`. I find as a minimum having overloads of `persistState` a bit confusing: is one `persistCopiedState` or something similar?